src/scan_validator.cpp

```diff
@@ -58,30 +59,43 @@ bool laser_filters::ScanValidator::update(
   // Copy input_scan data
   output_scan = input_scan;
 
-  int number_threshold = static_cast<int>(violation_percentage_ * input_scan.ranges.size());
-  int cur_count = 0;
+  int occlusion_threshold_num = static_cast<int>(occlusion_threshold_ * input_scan.ranges.size());
+  int zero_threshold_num = static_cast<int>(zero_threshold_ * input_scan.ranges.size());
+  int occlusion_count = 0;
+  int zero_count = 0;
 
   // Traverse each point
   for(auto i = 0; i < input_scan.ranges.size(); i++)
   {
+    // zero reading usually means obstacle is further than LIDAR max range
+    if(input_scan.ranges[i] == 0)
+    {
+      zero_count += 1;
+      continue;
+    }
     // Counts for the reading is NaN or smaller than contour range
     // Make the contour slightly smaller to avoid false positive case
     if(std::isnan(input_scan.ranges[i]) ||
        input_scan.ranges[i] < (contour_[i] - contour_tolerance_))
     {
-      cur_count += 1;
+      occlusion_count += 1;
     }
   }
 
-  int occlusionPercentage = static_cast<int>((static_cast<float>(cur_count) * 100) / static_cast<float>(input_scan.ranges.size()));
+  if(zero_count >= zero_threshold_num)
```

Comment:
We should add a flag to disable this filter and disable it by default.

Reply:
Do we need to do so if it simply split out a warn log?